### PR TITLE
Ps 5.7 myr 148

### PR DESCRIPTION
**Pull request by george-lorch**
Merge May 2017 changes

### CODE REVIEW

**Comment by george-lorch** [comment]
jenkins http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1164/

rocksdb.ttl_* and rocksdb.parts tests still going to be unstable as they only changed from sleep to real_sleep in mtr tests. Coming merges add code and debug variables for better synchronization and test stability.

File: mysql-test/suite/rocksdb/t/index_merge_rocksdb-master.opt [hunk 1]
```diff
@@ -0,0 +1 @@
+--loose-rocksdb_strict_collation_check=off --binlog_format=row --log-bin
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/index_merge_rocksdb-master.opt]: Why the last two?

File: mysql-test/suite/rocksdb/t/index_merge_rocksdb2-master.opt [hunk 2]
```diff
@@ -0,0 +1 @@
+--loose-rocksdb_strict_collation_check=off --binlog_format=row --log-bin --transaction-isolation='READ-COMMITTED'
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/index_merge_rocksdb2-master.opt]: Dynamic variables better set in the testcase directly

File: storage/rocksdb/rdb_sst_info.cc [hunk 3]
```diff
@@ -98,6 +98,8 @@ Rdb_sst_file_ordered::Rdb_sst_file::put(const rocksdb::Slice &key,
   DBUG_ASSERT(m_sst_file_writer != nullptr);
 
   // Add the specified key/value to the sst file writer
+#pragma GCC diagnostic push
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_sst_info.cc:101]: This is a .cc file and thus benign, but I'd prefer to have a matching pop in order for this to survive any future refactoring better

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: mysql-test/suite/rocksdb/t/index_merge_rocksdb-master.opt (hunk 1 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/index_merge_rocksdb-master.opt]: Dunno, came from upstream and I didn't really notice. Good question as binlog/rpl shouldn't have anything to do with this test.

File: mysql-test/suite/rocksdb/t/index_merge_rocksdb2-master.opt (hunk 2 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/index_merge_rocksdb2-master.opt]: true, binlog options again are probably not needed here, but, trx iso I think is as IIRC the test includes files that use multiple connections...I'll have to look to be sure.

File: storage/rocksdb/rdb_sst_info.cc (hunk 3 above)
Comment (george-lorch) [reply storage/rocksdb/rdb_sst_info.cc:101]: yeah, and as I understand this is supposed to be temporary. I'll look ahead in merge history to see if there is another change coming, if not, I'll add a pop.

File: mysql-test/suite/rocksdb/t/index_merge_rocksdb2-master.opt (hunk 2 above)
Comment (george-lorch) [reply mysql-test/suite/rocksdb/t/index_merge_rocksdb2-master.opt]: Actually, I see the problem, index_merge1.inc and possibly others restart mysqld, which wipes/resets any dynamic setting back to default. Thus iso needs to be set here in the .opt file to ensure it survives restarts.

**Comment by george-lorch** [comment]
Forgot the new jenkins link http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1166/

**Comment by george-lorch** [comment]
Going ahead and merging since everything was addressed and jenkins runs are clean.
Reported MYR-169 : rocksdb.sys_vars.rocksdb_pause_background_work_basic is unstable

**Pull request merged**